fms/models/mpnet.py

```diff
@@ -0,0 +1,480 @@
+            self.config.emb_dim // self.config.nheads,
+            self.config.nheads,
+            kvheads=1 if self.config.multiquery_attn else self.config.nheads,
+            p_dropout=self.config.hidden_dropout_prob,
+            use_bias=True,
+            fused=self.config.fused_weights,
+            linear_config=self.config.linear_config,
+        )
+        self.ln = nn.LayerNorm(self.config.emb_dim, self.config.layer_norm_eps)
+        self.ff_sub_layer = FeedForwardBlock(
+            self.config.emb_dim,
+            hidden_grow_factor=self.config.hidden_grow_factor,
+            activation_fn=str_to_activation(self.config.activation_fn),
+            p_dropout=self.config.hidden_dropout_prob,
+            use_bias=True,
+            linear_config=self.config.linear_config,
+        )
+        self.ff_ln = nn.LayerNorm(self.config.emb_dim, self.config.layer_norm_eps)
+        if self.config.hidden_dropout_prob != 0:
+            self.dropout = nn.Dropout(self.config.hidden_dropout_prob)
+      
+    def forward(
+        self,
+        x: torch.Tensor,
+        position_ids: torch.Tensor,
+        **attn_kwargs: Unpack[AttentionKwargs],
+    ):
+        residual = x
+        x = self.attn(
+            q=x,
+            **attn_kwargs,
+        )
+        #print("reached here")
+        if self.config.hidden_dropout_prob != 0:
+            x = self.dropout(x)
+        x = x + residual
+        residual = x
+        x = self.ln(x)
+        x = self.ff_sub_layer(x)
+        x = self.ff_ln(x)
+        if self.config.hidden_dropout_prob != 0:
+            x = self.dropout(x)
+        x = x + residual
+
+        return x
+
+
+class MpnetHeadless(nn.Module):
+    def __init__(
+        self,
+        config: Optional[MpnetConfig] = None,
+        **kwargs,
+    ):
+        super(MpnetHeadless, self).__init__()
+        if config is not None:
+            self.config = config
+        else:
+            self.config = MpnetConfig()
+        self.config = self.config.updated(**kwargs)
+        self.width = self.config.emb_dim
+        self.pad_id = self.config.pad_id
+        self.max_expected_seq_len = self.config.max_expected_seq_len
+
+        #self.embedding = nn.Embedding(
+        #    self.config.src_vocab_size,
+        #    self.config.emb_dim,
+        #    padding_idx=self.config.pad_id,
+        #)
+
+        self.embedding = nn.Embedding(config.src_vocab_size, config.emb_dim, padding_idx=self.pad_id)
+        self.position_embeddings = nn.Embedding(
+            config.max_expected_seq_len, config.emb_dim, padding_idx=self.pad_id
+        )
+
+        self.enc_norm = nn.LayerNorm(config.emb_dim, eps=config.layer_norm_eps)
+        self.dropout = nn.Dropout(config.hidden_dropout_prob)
+        self.register_buffer(
+            #"position_ids", torch.arange(config.max_expected_seq_len).expand((1, -1)), persistent=False
+            "position_ids", torch.arange(514).expand((1, -1)), persistent=False
+        )
```

Comment:
this will cause issues with devices, please take a look at siglip vision PR for how to do this properly so it works on cpu/gpu/aiu/etc.